helix-core/src/main/java/org/apache/helix/common/cloud/CloudInformationParser.java

```diff
@@ -0,0 +1,34 @@
+package org.apache.helix.common.cloud;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.List;
+
+
+/**
+ * Generic interface to parse and validate cloud information
+ */
+public interface CloudInformationParser {
+  /**
+   * Parse the cloud information from responses
+   * @return the parsed cloud information
+   */
+  CloudInformation parseCloudInformation(List<String> cloudResponses);
```

Comment:
For interface level, I think we can make it as generic type T. For implementation, like AzureCloudInformationParser, let's make it as String.